src/Datasource/QueryTrait.php

```diff
@@ -276,6 +276,10 @@ public function aliasFields(array $fields, ?string $defaultAlias = null): array
     public function all(): ResultSetInterface
     {
         if ($this->_results !== null) {
+            if (!$this->_results instanceof ResultSetInterface) {
+                $this->_results = $this->_decorateResults($this->_results);
```

Comment:
Will this decorate multiple times?

Reply:
The instanceof check ensures the results are decorated only once.

This is to ensure that calling `$query->all()` after calling `$query->setResult([..])` doesn't cause a type error since `setResult()` accepts any iterable while `all()` should always return a `ResultSetInterface`.

Reply:
Ah, my mistake. I did not see the `!` operator.